common/flatpak-dir.c

```diff
@@ -7139,6 +7139,8 @@ export_desktop_file (const char         *app,
                 g_string_append_printf (new_exec, " @@ %s @@", arg);
               else if (strcasecmp (arg, "%u") == 0)
                 g_string_append_printf (new_exec, " @@u %s @@", arg);
+              else if (strcmp (arg, "@@") == 0 || strcmp (arg, "@@u") == 0)
+                g_print (_("Skipping invalid Exec argument %s\n"), arg);
```

Comment:
Since this is either an attempted attack or something that will break `%u` and `%f` processing, I'd be tempted to make it fail hard:

```suggestion
                {
                  flatpak_fail_error (error, FLATPAK_ERROR_EXPORT_FAILED,
                                     _("Invalid Exec argument %s"), arg);
                  goto out;
                }
```

Reply:
yes, agreed

Reply:
#4156 incorporates this suggestion.